src/lucky_cli/task.cr

```diff
@@ -60,6 +68,129 @@ abstract class LuckyCli::Task
     end
   end
 
+  # Creates a method of `arg_name` that returns the value passed in from the CLI.
+  # The CLI arg position is based on the order in which `positional_arg` is specified
+  # with the first call being position 0, and so on.
+  #
+  # If your arg takes more than one value, you can set `to_end` to true to capture all
+  # args from this position to the end. This will make your `arg_name` method return `Array(String)`.
+  #
+  # * `arg_name` : String - The name of the argument
+  # * `description` : String - The help text description for this option
+  # * `required` : Bool - raise exception if this arg is not passed
+  # * `to_end` : Bool - Capture all args from this position to the end.
+  # * `format` : Regex - The format you expect the args to match
+  macro positional_arg(arg_name, description, required = false, to_end = false, format = nil)
+    {% PARSER_OPTS << arg_name %}
+    @{{ arg_name.id }} : {% if to_end %}Array(String){% else %}String{% end %} | Nil
+
+    def set_opt_for_{{ arg_name.id }}(args : Array(String))
+      {% if to_end %}
+        value = args[@positional_arg_count..-1]
+      {% else %}
+        value = args[@positional_arg_count]?
+      {% end %}
+      {% if format %}
+      matches = value.is_a?(Array) ? value.all?(&.=~({{ format }})) : value =~ {{ format }}
+      if matches
+        @{{ arg_name.id }} = value
+      else
+        raise "Invalid format for {{ arg_name.id }}. It should match {{ format }}"
+      end
+      {% else %}
+        @{{ arg_name.id }} = value
+      {% end %}
+      @positional_arg_count += 1
+    end
+
+    def {{ arg_name.id }}
```

Comment:
I wanted to have the return type set, but there's 4 possible options here `Array(String)`, `Array(String)?` `String` and `String?` which makes the macro code really funky and difficult to read. I figure we can let the compiler handle it on this case.